packages/builder/src/wormhole/WormholeBuilder.ts

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import { Wormhole } from '@wormhole-foundation/sdk-connect';
+import { EvmParachain, Parachain } from '@moonbeam-network/xcm-types';
+import { stringify, stringToU8a } from '@polkadot/util';
+import { evmToAddress } from '@polkadot/util-crypto/address';
+import {
+  WormholeConfigBuilder,
+  WormholeConfigBuilderPrams,
+} from './WormholeBuilder.interfaces';
+import { WormholeConfig } from './WormholeConfig';
+import { wormholeFactory } from './wormholeFactory';
+import { getExtrinsicAccount } from '../extrinsic/ExtrinsicBuilder.utils';
+
+export const GMP_CONTRACT_ADDRESS =
+  '0x0000000000000000000000000000000000000816';
+
+export function WormholeBuilder() {
+  return {
+    tokenTransfer: ({
+      isAutomatic,
+    }: {
+      isAutomatic: boolean;
+    }): WormholeConfigBuilder => ({
+      build: ({
+        asset,
+        destination,
+        destinationAddress,
+        moonChain,
+        source,
+        sourceAddress,
+      }) => {
+        if (!asset.address) {
+          throw new Error(`Asset ${asset.key} has no address`);
+        }
+
+        const isDestinationMoonChain = destination.isEqual(moonChain);
+
+        const wh = wormholeFactory(source);
+        const whSource = wh.getChain(source.getWormholeName());
+        const whMoonChain = wh.getChain(moonChain.getWormholeName());
+        const whAsset = Wormhole.tokenId(whSource.chain, asset.address);
+        const whSourceAddress = Wormhole.chainAddress(
+          whSource.chain,
+          sourceAddress,
+        );
+        const whMoonChainAddress = Wormhole.chainAddress(
+          whMoonChain.chain,
+          isDestinationMoonChain ? destinationAddress : GMP_CONTRACT_ADDRESS,
+        );
+
+        return new WormholeConfig({
```

Comment:
I don't quite get it how this will be used.
Because as I understand here, it will be like fantom to peaq transfer. I guess we have to create here wormhole sdk transfer here?

Reply:
It will be used the same way as other builders in the SDK. Here I'm building config for a transfer. Which I will do in the later stage in `mrl` package.

Reply:
ok